pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -875,6 +884,29 @@ func (b *Bond) PeerNotifyDelay() uint32 {
 	return b.BondPeerNotifyDelay
 }
 
+// Enabled implements the config.STP interface.
+func (s *STP) Enabled() bool {
+	if s.STPEnabled == nil {
```

Comment:
we can also check for `nil` here, and simplify places where we read this value:

```suggestion
	if s == nil || s.STPEnabled == nil {
```

so that we don't have to check for `STP()` being nil, and do `device.STP().Enabled()` always

Reply:
makes sense, changed